terraform/cloudsql/postgresql/provision/data.tf

```diff
@@ -4,4 +4,7 @@ data "google_compute_network" "authorized-network" {
 
 locals {
   authorized_network_id = length(var.authorized_network_id) > 0 ? var.authorized_network_id : data.google_compute_network.authorized-network.self_link
+  availability_type     = var.highly_available ? "REGIONAL" : "ZONAL"
+  primary_zone          = var.location_preference_zone == "" ? "" : join("-", [var.region, var.location_preference_zone])
```

Comment:
Andrea and I discussed the pros and cons of this implementation.
The biggest pro he shared, in my opinion, is that MySQL is implemented in the exact same way.
I'm ok going with this implementation for consistency. We can decide to refactor both implementations in the future if needed (although it is very unlikely we will do so because it would imply forcing customer to change their plans and instances).